### PR TITLE
updates for v4.0 and minor tweaks

### PR DESCRIPTION
**Pull request by rvagg**
Put a versions.json in here for v4 and couldn't help but fix up a bunch of things:

* put the name in the downloads list, Node.js or io.js
* put a correct download URL
* change a whole lot of "node.js" to "Node.js", sorry, I've got a beef with this one, does anyone object? @mikeal, @fhemberger?
* change NPM to npm in the header of the downloads table

### CODE REVIEW

**Comment by fhemberger** [comment]
LGTM :+1:

**Comment by rvagg** [comment]
Added some text to a release post, please review @mikeal and others, @nodejs/tsc perhaps? See the big locale/en/blog/release/v4.0.0.md  file in this PR.

**Comment by jasnell** [comment]
Will review shortly. 10-15 minutes or so

File: layouts/download-releases.hbs [hunk 1]
```diff
@@ -24,19 +24,19 @@
...
                             </tr>
                         </thead>
                         <tbody>
                             {{#project.versions}}
                                 <tr>
-                                    <td data-label="Version">{{version}}</td>
+                                    <td data-label="Version">{{name}} {{version}}</td>
                                     <td data-label="Date"><time>{{date}}</time></td>
                                     <td data-label="V8">{{v8}}</td>
                                     <td data-label="NPM">{{npm}}</td>
```
Comment (evanlucas) [inline layouts/download-releases.hbs]: Does it matter that the `data-label` is still `NPM` instead of `npm`?

Comment (rvagg) [inline layouts/download-releases.hbs]: seems to work, I just didn't dare touch it because I don't know what it's doing

**Comment by rvagg** [comment]
you have time, I'm rerunning tests and making builds and whatnot since the late addition of the handle zapping change

**Comment by rvagg** [comment]
Added a new section on build/platforms/arm:

> Over the past year thanks to donations from companies such as Rackspace, DigitalOcean, Scaleway, miniNodes, Voxer, ARM Holdings, and individuals from the community, we've built an amazing test cluster that includes an impressive line-up of Linux variants, OS X, Windows, FreeBSD and SmartOS. New to Node.js, we also have first-class support for ARM processors. Our test suite must be continually passing on ARMv6, ARMv7 and the new 64-bit ARMv8 processors. So Node.js is officially ready for use by hobbyists and ARM server users alike.

File: layouts/download-releases.hbs (hunk 1 above)
Comment (fhemberger) [inline layouts/download-releases.hbs]: You can change that as well, it's just the label for the mobile table version.

**Comment by fhemberger** [comment]
Text looks good as well.

**Comment by jasnell** [comment]
Looks good on initial read through. Nothing major sticks out.

**Comment by cjihrig** [comment]
+1 to what @jasnell said.

File: locale/en/blog/release/v4.0.0.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+Node.js v4.0.0 contains V8 v4.5, the same version of V8 shipping with the Chrome web browser today. This brings with it many bonuses for Node.js users, most notably a raft of new [ES6](https://nodejs.org/en/docs/es6/) features that are enabled by default including block scoping, classes, typed arrays, generators, Promises, Symbols, template strings, collections (Map, Set, etc.) and, new to V8 v4.5, arrow functions.
+
+Over the past year thanks to donations from companies such as Rackspace, DigitalOcean, Scaleway, miniNodes, Voxer, ARM Holdings, and individuals from the community, we've built an amazing test cluster that includes an impressive line-up of Linux variants, OS X, Windows, FreeBSD and SmartOS. New to Node.js, we also have first-class support for ARM processors. Our test suite must be continually passing on ARMv6, ARMv7 and the new 64-bit ARMv8 processors. So Node.js is officially ready for use by hobbyists and ARM server users alike.
```
Comment (mikeal) [inline locale/en/blog/release/v4.0.0.md]: Don't forget Cloudflare.

File: locale/en/blog/release/v4.0.0.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+As always with a major release, we don't expect the adoption of Node.js v4 to be painless. It will take time to adapt, particularly as compiled add-ons catch up with the new version of V8 (hint: see [NAN](https://github.com/nodejs/nan) for our official solution to the compatibility problem). Thankfully, the io.js v3.x series of releases have given many add-on authors a chance to prepare their code for Node.js v4, so much of the npm ecosystem is ready.
```
Comment (mikeal) [inline locale/en/blog/release/v4.0.0.md]: > npm ecosystem is ready

"npm ecosystem is ready today."

File: locale/en/blog/release/v4.0.0.md (hunk 2 above)
Comment (rvagg) [inline locale/en/blog/release/v4.0.0.md]: whoops, and Joyent, added

**Comment by mikeal** [comment]
Comments inline. Other than that, +1

File: source/versions.json [hunk 4]
```diff
@@ -1,74 +1,2827 @@
...
+    "uv": "0.10.18",
+    "zlib": "1.2.3",
+    "openssl": "1.0.1e",
+    "modules": "11",
+    "url": "https://nodejs.org/dist/v0.10.21/",
+    "name": "Node.js"
+  },
+  {
+    "version": "v0.10.20",
+    "date":
```
Comment (Fishrock123) [inline source/versions.json]: Can we just keep like a max of 3 releases in here? This file doesn't need to have them all when checked in to git.

File: locale/en/blog/release/v4.0.0.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
+---
+date: 2015-09-08T12:10:06.000Z
+version: 4.0.0
+category: release
+title: Node v4.0.0 (Stable)
+slug: node-v4-0-0-stable
+layout: blog-post.hbs
+---
+
+The collaborators of the Node.js project and the members of the Node.js Foundation are proud to offer v4.0.0 for general release. This release represents countless hours of hard work encapsulated in both the Node.js project and the io.js project that are now combined in a single codebase. The Node.js project is now operated by a team of 29 collaborators with additional 15 members forming its Technical Steering Committee. Further, over 100 new individuals have been added to the list of people contributing code to core since v0.12.7.
```
Comment (Fishrock123) [inline locale/en/blog/release/v4.0.0.md]: Prefer "44 collaborators, 15 of which are TSC members".

File: locale/en/blog/release/v4.0.0.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+
+Node.js v4.0.0 contains V8 v4.5, the same version of V8 shipping with the Chrome web browser today. This brings with it many bonuses for Node.js users, most notably a raft of new [ES6](https://nodejs.org/en/docs/es6/) features that are enabled by default including block scoping, classes, typed arrays, generators, Promises, Symbols, template strings, collections (Map, Set, etc.) and, new to V8 v4.5, arrow functions.
+
+Over the past year thanks to donations from companies such as Rackspace, DigitalOcean, Scaleway, miniNodes, Voxer, Joyent, CloudFlare, ARM Holdings, and individuals from the community, we've built an amazing test cluster that includes an impressive line-up of Linux variants, OS X, Windows, FreeBSD and SmartOS. New to Node.js, we also have first-class support for ARM processors. Our test suite must be continually passing on ARMv6, ARMv7 and the new 64-bit ARMv8 processors. So Node.js is officially ready for use by hobbyists and ARM server users alike.
```
Comment (Fishrock123) [inline locale/en/blog/release/v4.0.0.md]: Note that the ARM cluster is mostly community donated?

File: locale/en/blog/release/v4.0.0.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+version: 4.0.0
+category: release
+title: Node v4.0.0 (Stable)
+slug: node-v4-0-0-stable
+layout: blog-post.hbs
+---
+
+The collaborators of the Node.js project and the members of the Node.js Foundation are proud to offer v4.0.0 for general release. This release represents countless hours of hard work encapsulated in both the Node.js project and the io.js project that are now combined in a single codebase. The Node.js project is now operated by a team of 29 collaborators with additional 15 members forming its Technical Steering Committee. Further, over 100 new individuals have been added to the list of people contributing code to core since v0.12.7.
+
+Node.js v4.0.0 contains V8 v4.5, the same version of V8 shipping with the Chrome web browser today. This brings with it many bonuses for Node.js users, most notably a raft of new [ES6](https://nodejs.org/en/docs/es6/) features that are enabled by default including block scoping, classes, typed arrays, generators, Promises, Symbols, template strings, collections (Map, Set, etc.) and, new to V8 v4.5, arrow functions.
```
Comment (Fishrock123) [inline locale/en/blog/release/v4.0.0.md]: Should we note that buffers are now backed by the new Uint8Arrays?

File: locale/en/blog/release/v4.0.0.md (hunk 6 above)
Comment (rvagg) [inline locale/en/blog/release/v4.0.0.md]: it's about 50/50 when taken as a whole, and I mentioned community members above

File: locale/en/blog/release/v4.0.0.md [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+
+In parallel, we will be releasing a new Stable line of releases every 6 months, one in October and one in April each year. Every second one of these, the April release, will continue life in LTS form by the time October comes around. This means that there will be overlapping LTS branches being maintained throughout the year, each receiving attention for a total of 30 months (LTS plus Maintenance). We believe that the Stable and LTS cycles provide the room for the project to continue its **innovation** and stay with the cutting-edge of JavaScript while also serving the need for **predictable long-term stability and security** demanded by the growing line-up of enterprise users proudly adopting Node.js as a key technology; some of who already count themselves as [members of the Node.js Foundation](https://nodejs.org/en/foundation/members/).
```
Comment (Fishrock123) [inline locale/en/blog/release/v4.0.0.md]: Perhaps mention here or above that the stable line will have basically weekly-ish patch/minor releases?

File: source/versions.json (hunk 4 above)
Comment (fhemberger) [inline source/versions.json]: We need all release for the release history page: https://nodejs.org/en/download/releases/

File: locale/en/blog/release/v4.0.0.md [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+
+Other release files: http://nodejs.org/dist/v4.0.0/
+
+Website: http://nodejs.org/docs/v4.0.0/
+
+Documentation: http://nodejs.org/docs/v4.0.0/api/
+
+Shasums:
+```
+{{shasums}}
```
Comment (phillipj) [inline locale/en/blog/release/v4.0.0.md]: The list of release files and their corresponding shasums are missing, e.g. for v3.3.0: https://iojs.org/dist/v3.3.0/SHASUMS256.txt.asc

For all I know you havent done the final build yet?

**Pull request merged**